File: framework/languages/node/e2e/ResilienceLifecycle/Server/TopologyProbe/Configuration/server-options.ts
import { objectValues, optionalString } from '../../../configuration';

export interface ServerOptions {
  readonly role: string;
  readonly httpUrl: string;
  readonly logDir: string;
  readonly evidenceFile?: string;
  readonly rid: string;
  readonly redisEndpoint: string;
  readonly redisKeyPrefix: string;
}

export function validateServerOptions(value: unknown, defaultRole = 'topology-probe'): ServerOptions {
  const values = objectValues(value);
  const redisEndpoint = optionalString(values, 'redisEndpoint');
  const redisKeyPrefix = optionalString(values, 'redisKeyPrefix');
  if (!redisEndpoint || !redisKeyPrefix) throw new Error("Configuration requires 'e2e.redisEndpoint' and 'e2e.redisKeyPrefix'.");
  return {
    role: defaultRole,
    httpUrl: optionalString(values, 'httpUrl') ?? 'http://127.0.0.1:0',
    logDir: optionalString(values, 'logDir') ?? '/tmp/zlink-node-e2e-log',
    evidenceFile: optionalString(values, 'evidenceFile'),
    rid: optionalString(values, 'rid') ?? 'node',
    redisEndpoint,
    redisKeyPrefix
  };
}

File: framework/languages/dotnet/e2e/LocationMessaging/Server/Provider/Configuration/ServerOptions.cs
namespace LocationMessaging.Server.Provider.Configuration;

using Zlink.Framework.E2E.Configuration;

internal sealed record ServerOptions(
    string Role,
    string HttpUrl,
    string LogDir,
    string? EvidenceFile,
    string Rid,
    string? RedisEndpoint,
    string? RedisKeyPrefix,
    string? ChannelEndpoint,
    string? ManualClientEndpoint,
    string? RouteEndpoint,
    int Weight,
    long MaxMessageSize,
    IReadOnlyList<string>? RoutePeers = null)
{
    public static ServerOptions Parse(string[] args, string defaultRole)
        => E2eConfiguration.Load<ServerOptions>(args) with { Role = defaultRole };
}
